app/controllers/documents/document_upload_question_controller.rb

```diff
@@ -21,9 +21,11 @@ def update
         form_saved = @form.save
         after_update_success
         track_document_upload
+        redirect_to action: :edit
+      else
+        track_validation_error
+        render :edit
```

Comment:
this follows the same pattern as the questions controller - would love context on why we weren't doing this in the first place and whether there's a problem with it. the reason i changed it was that rendering errors in the usual way doesn't work if you redirect